### PR TITLE
[FEATURE]Added cancel button on search bar when platform is iOS

### PR DESCRIPTION
**Pull request by juanpetterson**
@RocketChat/ReactNative

Added cancel button on search bar when platform is iOS, based on the iOS design guidelines.

### CODE REVIEW

File: app/containers/SearchBox.js [hunk 1]
```diff
@@ -25,7 +25,10 @@ const styles = StyleSheet.create({
 		flexDirection: 'row',
 		fontSize: 17,
 		height: 36,
-		margin: 16,
+		marginBottom: 16,
+		marginRight: 15,
+		marginTop: 16,
+		marginLeft: 16,
 		marginVertical: 10,
```
Comment (pranavpandey1998official) [inline app/containers/SearchBox.js]: you either have to remove marginBottom marginTop or marginVertical

File: app/containers/SearchBox.js [hunk 2]
```diff
@@ -25,7 +25,10 @@ const styles = StyleSheet.create({
 		flexDirection: 'row',
 		fontSize: 17,
 		height: 36,
-		margin: 16,
+		marginBottom: 16,
```
Comment (diegolmello) [inline app/containers/SearchBox.js]: How does this affects other places using SearchBox?

File: app/containers/SearchBox.js [hunk 3]
```diff
@@ -40,7 +43,7 @@ const styles = StyleSheet.create({
 		...sharedStyles.textRegular
 	},
 	cancel: {
-		marginRight: 10
+		marginRight: 14
```
Comment (diegolmello) [inline app/containers/SearchBox.js]: How does this affects other places using SearchBox?

File: app/views/RoomsListView/index.js [hunk 4]
```diff
@@ -424,6 +425,10 @@ class RoomsListView extends React.Component {
 
 	getScrollRef = ref => this.scroll = ref
 
+	onCancelPress = () => {
+		this.cancelSearchingAndroid();
```
Comment (diegolmello) [inline app/views/RoomsListView/index.js]: I think we should rename this function. It's not Android only anymore.

File: app/containers/SearchBox.js [hunk 5]
```diff
@@ -25,7 +25,10 @@ const styles = StyleSheet.create({
 		flexDirection: 'row',
 		fontSize: 17,
 		height: 36,
-		margin: 16,
+		marginBottom: 16,
+		marginRight: 15,
```
Comment (diegolmello) [inline app/containers/SearchBox.js]: Why 15 here and 16 on others?
This is anti-pattern.

**Comment by juanpetterson** [comment]
Closed the PR because the [#1127](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/1127) was opened.

**Pull request closed without merging**